src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -336,3 +338,152 @@ Json::Value Profile::ToJson() const
 
     return json;
 }
+winrt::hstring Profile::EvaluatedIcon()
+{
+    // We cache the result here, so we don't search the path for the exe every time.
+    if (!_evaluatedIcon.has_value())
+    {
+        _evaluatedIcon = _evaluateIcon();
+    }
+    return *_evaluatedIcon;
+}
+
+winrt::hstring Profile::_evaluateIcon() const
+{
+    // If the profile has an icon, return it.
+    if (!Icon().empty())

```

Comment:
can we not use our own member here?

Reply:
Nah, we can't because profile is layered so `_Icon` isn't the actual icon necessarily